webpack.config.js

```diff
@@ -1,17 +1,79 @@
 const path = require("path");
 const webpack = require('webpack');
+const HTMLWebpackPlugin = require('html-webpack-plugin');
+const CompressionPlugin = require("compression-webpack-plugin");
+const SWPrecacheWebpackPlugin = require('sw-precache-webpack-plugin');
 
-const config = {
-  entry: {
-    app: [
-      './src/index.js'
-    ]
-  },
 
+
+
+//enviroment variable
+const DEVELOPMENT = process.env.NODE_ENV === 'development';
+const PRODUCTION = process.env.NODE_ENV === 'production';
+const TEST = process.env.NODE_ENV === 'test'
+
+const entry = PRODUCTION
+      ?     {
+              src:
+              [
+                './index.js'
+              ]
+            }
+      :
+              [
+                './index.js'
+              ];
+
+const plugins = PRODUCTION
```

Comment:
I think there might be a better way to write this part of the code, switching between environments? What if you had this code like:
```javascript
const plugins = {};
plugins.production = [ ... ];
plugins.development = [new webpack.NamedModulesPlugin()];
plugins.test = plugins.development // or whatever

// then when using the plugin, just use
plugins[process.env.NODE_ENV
```